mm2src/mm2_main/src/lp_ordermatch.rs

```diff
@@ -2734,6 +2739,20 @@ impl OrdermatchContext {
     pub async fn ordermatch_db(&self) -> InitDbResult<OrdermatchDbLocked<'_>> {
         Ok(self.ordermatch_db.get_or_initialize().await?)
     }
+
+    fn add_pubkey(&self, p2p_privkey: &Option<SerializableSecp256k1Keypair>) {
+        if let Some(p2p_privkey) = p2p_privkey {
+            let pubsecp = hex::encode(p2p_privkey.public_slice());
+            self.my_p2p_pubkeys.lock().insert(pubsecp);
+        }
+    }
+
+    fn remove_pubkey(&self, p2p_privkey: &Option<SerializableSecp256k1Keypair>) {
```

Comment:
Do we need to use `remove_pubkey` at the following places?
[cancel_orders_by 1](https://github.com/KomodoPlatform/atomicDEX-API/blob/ZHTLC_orders_ismine_bug/mm2src/mm2_main/src/lp_ordermatch.rs#L5248), [cancel_orders_by 2](https://github.com/KomodoPlatform/atomicDEX-API/blob/ZHTLC_orders_ismine_bug/mm2src/mm2_main/src/lp_ordermatch.rs#L5266), [cancel_orders_by 3](https://github.com/KomodoPlatform/atomicDEX-API/blob/ZHTLC_orders_ismine_bug/mm2src/mm2_main/src/lp_ordermatch.rs#L5286)

Anyway, it's not critical at all since it's better to forget to remove than to add a pubkey.

Reply:
Yeah, I've got what you mean. I had this implementation later and removed in commit named [remove unnecessary code ](https://github.com/KomodoPlatform/atomicDEX-API/commit/112a20782129b8d01404c252da3a2feabe38aaa0#diff-4c4940ed0030a5a26cc593870df06f290339e4ed69d0e7564b2c71135e4d2787R5246-R5248).
I got this [note](https://github.com/KomodoPlatform/atomicDEX-API/pull/1489#discussion_r993466525) by Artem.
We can remove orders later int iteration over [cancelled_maker_orders](https://github.com/KomodoPlatform/atomicDEX-API/commit/112a20782129b8d01404c252da3a2feabe38aaa0#diff-4c4940ed0030a5a26cc593870df06f290339e4ed69d0e7564b2c71135e4d2787R5296-R5297). Bcz before this iteration in match we use [macro](https://github.com/KomodoPlatform/atomicDEX-API/blob/ZHTLC_orders_ismine_bug/mm2src/mm2_main/src/lp_ordermatch.rs#L5202) where we push orders in vector `cancelled_maker_orders`. I didnt notice it, so did unnecessary implementation.
So we dont forget to delete orders, we just do it later in one iteration.

Reply:
Got it, thank you!